pkg/controller/podgc/gc_controller.go

```diff
@@ -112,19 +114,19 @@ func (gcc *PodGCController) Run(ctx context.Context) {
 	<-ctx.Done()
 }
 
-func (gcc *PodGCController) gc(ctx context.Context) {
+func (gcc *PodGCController) gc(logger klog.Logger, ctx context.Context) {
```

Comment:
I believe logger can be retrived from context `logger := klog.FromContext(ctx)`
not sure why you've taken `logger` as input, similar for other funcs